README.md

```diff
@@ -50,14 +50,9 @@ To learn more about the programming language, visit [swift.org](https://swift.or
 
 - [Contributing to Swift](#contributing-to-swift)
 - [Getting Started](#getting-started)
-  - [System Requirements](#system-requirements)
-  - [Getting Sources for Swift and Related Projects](#getting-sources-for-swift-and-related-projects)
-  - [Building Swift](#building-swift)
   - [Swift Toolchains](#swift-toolchains)
   - [Build Failures](#build-failures)
-- [Testing Swift](#testing-swift)
 - [Learning More](#learning-more)
-- [Build Dependencies](#build-dependencies)
```

Comment:
Do the *Swift Toolchains* and *Build Failures* sections still belong in README.md, or should they be moved into GettingStarted.md?

Could the *Building Documentation* section be moved into FAQ.md (or removed, if the plan is to convert all `.rst` files to Markdown)?

Reply:
I do think that the README is not the right place for this information.

- Swift Toolchains section: I think a "How do I build and use a toolchain" guide would be valuable in its own right, so smooshing it together in the Getting Started guide is not a good idea, even though they share some parts in common. Instead, I think it would be better to create a separate document, and reference parts of the Getting Started guide as applicable.
  I just haven't had the time to look at this myself.
- Build Failures section: There are certain parts of it which only make sense in the context of upgrading Xcode versions, which is not something that happens often. I don't think it makes sense to have those instructions be a part of the Getting Started guide. Moreover, the cost of those recommendations is high; doing `--clean` or `--reconfigure` ends up leading to large build times. Admittedly, Sccache ameliorates this problem to an extent, but CMake is slow as sludge so...
   I would prefer that we have a separate explanatory document that dives more into what `build-script` and CMake are doing, and then talk about `--clean` and `--reconfigure`. As an example of what I'm talking about, I've been working on a [Compiler Architecture](https://github.com/varungandhi-apple/swift/blob/vg-all-new-docs/docs/Explanations/CompilerArchitecture.md) explanatory document.

OK. So I've kinda' answered the question from a ~medium term perspective. That still doesn't provide an immediate term answer: should we do _something_ for now, even if we're going to change it later?

I think my answer to that is that I can setup outline/sketch documents for what I talked about. That way, if someone is interested in fleshing those out, they have some scaffolding to work with and roughly know what is/isn't expected. Of course, the outline isn't meant to serve as a binding "plan of action", but as a guideline and an opportunity for discussion.

I do think that would require more work than simply moving a section from one place to another, so I'm inclined to setup the outline documents in a separate follow-up PR.

---

> Could the Building Documentation section be moved into FAQ.md (or removed, if the plan is to convert all .rst files to Markdown)?

Sphinx does support Markdown, so I suspect that part isn't necessarily a blocker. The main question I have is: is there interest in reading the standalone documentation (not talking about doc comments in code) as locally generated static HTML instead of reading it rendered on GitHub or inside a text editor as ReST/Markdown. At some point last year, I remember trying to build the documentation, and encountering oddball Python errors. I didn't have the patience to see it through.

I think moving it into the FAQ is a reasonable immediate-term "cleanup" suggestion that I can incorporate in this PR. I'm going to try the instructions and if they don't work, I'll add a disclaimer or something.

Reply:
> I would prefer that we have a separate explanatory document that dives more into what `build-script` and CMake are doing, and then talk about `--clean` and `--reconfigure`.

My experience was that `--reconfigure` was necessary when adding `--ios` to an incremental build.

> As an example of what I'm talking about, I've been working on a [Compiler Architecture](https://github.com/varungandhi-apple/swift/blob/vg-all-new-docs/docs/Explanations/CompilerArchitecture.md) explanatory document.

(Typo in title:)

```diff
-# Compiler Architecuture
+# Compiler Architecture
```

Reply:
> My experience was that --reconfigure was necessary when adding --ios to an incremental build.

Right... so the mental model is that `--reconfigure` should be used when the configuration changed. But in practice, it is not really clear what _exactly_ does/does not fall in this bucket of "configuration changed". As a result, you can go both ways -- try to be too optimistic and end up seeing issues that no one else has seen (alternatively you can fall-back to clean-build the minute you see a build error, but then you just wasted a build), try to be too pessimistic and spend up too much time clean-building things too much.

I suspect this would benefit from some actual measurements on build times. Is the additional complexity/nuance worth it. I don't know for sure. I suspect it is, but I'd have to measure.